### ISSUE TITLE
No pixels defined in cache `[0]' @ error/cache.c/SetPixelCacheNexusPixels/5132

### ISSUE THREAD

**Issue by jeroen** [ImageMagick/ImageMagick6#46]
After [updating to 6.9.10-42](https://github.com/Homebrew/homebrew-core/commit/70156a28cc1d74275d8ac68fb432eb158a9d80ab#diff-ddfdc4d33c234f38a2633ed22fdd4db2) the R bindings (via C++) give this error when calling `writeImages()` with a `std::vector<Magick::Image>` with more than 1 Image in it:

```
No pixels defined in cache `[0]' @ error/cache.c/SetPixelCacheNexusPixels/5132
```

It seems to happen for graphics when clipping is enabled. It does not happen when I disable clipping.

It also does not happen if the `std::vector<Magick::Image>` only has a single Image. Any guess what causes this? If needed I can try to produce a C++ example...

**Comment by urban-warrior** [comment]
Please post a simple C++ example that illustrates the problem.  Once we can reproduce the problem, we will have a patch to fix the problem, likely within a day.  Thanks.

**Comment by jeroen** [comment]
OK here is an example that sort of mimics my graphics system (I haven't cleaned it up yet):


```cpp
#include <Magick++.h>

typedef std::list<Magick::VPath> pathlist;
typedef std::list<Magick::Drawable> drawlist;

void set_clip(Magick::Image *ptr){
  pathlist path;
  path.push_back(Magick::PathMovetoAbs(Magick::Coordinate(0, 0)));
  path.push_back(Magick::PathLinetoAbs(Magick::Coordinate(600, 0)));
  path.push_back(Magick::PathLinetoAbs(Magick::Coordinate(600, 400)));
  path.push_back(Magick::PathLinetoAbs(Magick::Coordinate(0, 400)));
  path.push_back(Magick::PathLinetoAbs(Magick::Coordinate(0, 0)));

  drawlist draw;
  std::string id("mypath");
  draw.push_back(Magick::DrawablePushClipPath(id));
  draw.push_back(Magick::DrawablePath(path));
  draw.push_back(Magick::DrawablePopClipPath());
  draw.push_back(Magick::DrawableClipPath(id));

  ptr->draw(draw);
}

int main(){
  Magick::Blob blob;
  std::vector<Magick::Image> *vec = new std::vector<Magick::Image>();
  for(int i = 0; i < 5; i++){
    Magick::Image x(Magick::Geometry(600, 400), Magick::Color("red"));
    x.magick("PNG");
    x.depth(8L);
    set_clip(&x);
    vec->push_back(x);
    set_clip(&x);
  }

  for_each ( vec->begin(), vec->end(), Magick::gifDisposeMethodImage(MagickCore::BackgroundDispose));
  std::vector<Magick::Image> *output = new std::vector<Magick::Image>();
  coalesceImages( output, vec->begin(), vec->end());
  for_each ( output->begin(), output->end(), Magick::magickImage(std::string("gif")));
  writeImages( output->begin(), output->end(),  &blob );  
}

```

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by jeroen** [comment]
Thank you

**Comment by jeroen** [comment]
This bug is now back in version 6.9.10-49:

```
 Error: processing vignette 'intro.Rmd' failed with diagnostics:
 Magick: no pixels defined in cache `[0]' @ error/cache.c/SetPixelCacheNexusPixels/5154
```

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ http://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by jeroen** [comment]
Fixed again in 6.9.10-50. Thank you.
